cabal.project

```diff
@@ -1,6 +1,8 @@
 packages: .
 -- tests: True
 
+with-compiler: ghc-8.2.2
```

Comment:
please drop this

Reply:
Sure thing. I'm curious, can you explain why? The first thing I hit when `cabal new-build`ing was a base range error because I had `8.4.2` globally installed. I thought that's what the field is for - selecting a suitable `-w VER` implicitly.

Reply:
@mitchellwrosen because it would break for everyone not having ghc-8.2.2 installed even though they might have a different version of GHC installed which works perfectly. I consider it an anti-pattern to pin down the ghc version used in the cabal.project file pinned into a repository, *unless* that's the only supported GHC version; this would be as silly as checking in a freeze file or a Stack.yaml file.

Reply:
@hvr Ok, I've removed the line. For the record, I don't consider checking a freeze file silly, I do it with all of my projects. Solve -> Freeze -> Don't solve again until I have a new dependency to integrate. _Shrug_